libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/PolicyManager.java

```diff
@@ -28,6 +31,14 @@
 public class PolicyManager {
     private static final Logger logger = LogManager.getLogger(ElasticsearchEntitlementChecker.class);
 
+    static class ModuleEntitlements {
+        public static final ModuleEntitlements NONE = new ModuleEntitlements();
+        final List<FlagEntitlement> flagEntitlements = new ArrayList<>();
```

Comment:
>I'm still not convinced we need "flag" entitlements, and this makes me think so even more.

Absolutely, this is temporary and the reason I started introducing the CreateClassLoader entitlement. They will go away soon.
Then, my idea was that this list will be replaced by a single instance of each type (e.g. CreateClassLoader), but as they have no argument it can be a boolean or a map as you suggested.
I think we'll have 10 or so of them, so neither is a problem (we can have 10 fields and be super fast, or a map of 10s elements and still be super fast :) )

> For files we'll want a smarter object to do path matching, but that's a separate issue

Agreed; that was a reason to keep them as separate fields, so we can call specific methods on them. But we can do pattern-type matching, or have a different path for those, no prob. But yes, that is for another PR.

Reply:
Now that CreateClassLoader is merged, I changed this to use specific types and a IdentityHashMap.
It can still be optimized (atm I'm creating a "multi-map", we can consider merging/optimizing entitlements upon creation), but that will be for a follow-up